lib/matplotlib/patheffects.py

```diff
@@ -373,3 +375,148 @@ def draw_path(self, renderer, gc, tpath, affine, rgbFace):
         if clip_path:
             self.patch.set_clip_path(*clip_path)
         self.patch.draw(renderer)
+
+
+class TickedStroke(AbstractPathEffect):
+    """
+    A line-based PathEffect which draws a path with a ticked style.
+
+    This line style is frequently used to represent constraints in
+    optimization.  The ticks may be used to indicate that one side
+    of the line is invalid or to represent a closed boundary of a
+    domain (i.e. a wall or the edge of a pipe).
+
+    The spacing, length, and angle of ticks can be controlled.
+
+    This line style is sometimes referred to as a hatched line.
+
+    See also the :doc:`contour demo example
+    </gallery/lines_bars_and_markers/lines_with_ticks_demo>`.
+
+    See also the :doc:`contours in optimization example
+    </gallery/images_contours_and_fields/contours_in_optimization_demo>`.
+    """
+
+    def __init__(self, offset=(0, 0),
+                 spacing=10.0, angle=45.0, length=np.sqrt(2),
+                 **kwargs):
+        """
+        Parameters
+        ----------
+        offset : pair of floats, default: (0, 0)
+            The offset to apply to the path, in points.
+        spacing : float, default: 10.0
+            The spacing between ticks in points.
+        angle : float, default: 45.0
+            The angle between the path and the tick in degrees.  The angle
+            is measured as if you were an ant walking along the curve, with
+            zero degrees pointing directly ahead, 90 to your left, -90
+            to your right, and 180 behind you.
+        length : float, default: 1.414
+            The length of the tick relative to spacing.
+            Recommend length = 1.414 (sqrt(2)) when angle=45, length=1.0
+            when angle=90 and length=2.0 when angle=60.
+        **kwargs
+            Extra keywords are stored and passed through to
+            :meth:`AbstractPathEffect._update_gc`.
+
+        Examples
+        --------
+        See :doc:`/gallery/misc/tickedstroke_demo`.
+        """
+        super().__init__(offset)
+
+        self._spacing = spacing
+        self._angle = angle
+        self._length = length
+        self._gc = kwargs
+
+    def draw_path(self, renderer, gc, tpath, affine, rgbFace):
+        """
+        Draw the path with updated gc.
+        """
+        # Do not modify the input! Use copy instead.
+        gc0 = renderer.new_gc()
+        gc0.copy_properties(gc)
+
+        gc0 = self._update_gc(gc0, self._gc)
+        trans = affine + self._offset_transform(renderer)
+
+        thetar = -np.radians(self._angle)
```

Comment:
What's the 'r' standing for?

Reply:
Just a little note that it is now an angle in radians.

Reply:
That's too cryptic for me :smile:. Either use `theta_rad` or just leave it out. I would slightly prefer the latter. On the line itself `theta = -np.radians` is already clear, and the variable is only used in the next statement.